File: MemeLord/MemeLord/DataObjects/Dto/CommentDto.cs
using System;
using System.Collections.Generic;

namespace MemeLord.DataObjects.Dto
{
    public class CommentDto
    {
        public string Avatar { get; set; }
        public string Username { get; set; }
        public int Rating { get; set; }
        public IList<CommentDto> Answers { get; set; }
        public DateTime CreationDate { get; set; }
        public string Text { get; set; }
        public int Id { get; set; }
    }
}

File: AngularApp/src/app/guest-site/dto/comment-dto.ts
export interface ICommentDto{
    avatar: string,
    username: string,
    rating: number,
    //masterCommentId: number,
    creationDate: Date,
    text: string,
    id: number
}

export class CommentDto implements ICommentDto {
    avatar: string;
    username: string;
    rating: number;
    //masterCommentId: number;
    creationDate: Date;
    text: string;
    id: number;

    constructor(avatar: string, username: string, rating: number, /*masterCommentId: number, */creationDate: Date, text: string, id: number){
        this.avatar = avatar;
        this.username = username;
        this.rating = rating;
        //this.masterCommentId = masterCommentId;
        this.creationDate = creationDate;
        this.text = text;
        this.id = id;
    }
}